cmd/snap-update-ns/utils.go

```diff
@@ -457,13 +458,20 @@ func planWritableMimic(dir, neededBy string) ([]*Change, error) {
 
 	var changes []*Change
 
+	// Stat the original directory to know which mode and ownership to
+	// replicate on top of the tmpfs we are about to create below.
+	var sb syscall.Stat_t
+	if err := sysLstat(dir, &sb); err != nil {
+		return nil, err
```

Comment:
How is this affected by #5126 and #5090? Should we error out here?

Reply:
Ideally it should not be, this is directly in the code path that has decided to make a mimic.
When we do the mimic we have a starting point that we just want to replicate in a writable way.
This is not related to bailing out (in attempting to create a mimic) that #5126 or in the rules when we choose to consider it #5090